### DISCUSSION TITLE
ERROR: unrecognized configuration parameter "enable_experimental_alter_column_type_general"

### DISCUSSION THREAD

**Discussion by danieltamas** [prisma/prisma#22663] [Q&A] [answered]
### Bug description

Hello. I have  a cockroachDB schema for Nakama
While doing `npx prisma db pull --schema prisma/cockroach.prisma` 
I get the following error: 
```ERROR: unrecognized configuration parameter "enable_experimental_alter_column_type_general"```

Is there a way to exclude unsupported features like these?
Thank you

### How to reproduce

1. Go to `/prisma`
2. Create `/prisma/cockroach.prisma`
2. Add a new schema with a `cockroach` provider in the datasource
3. Run `npx prisma db pull --schema prisma/cockroach.prisma`
4. See error: `ERROR: unrecognized configuration parameter "enable_experimental_alter_column_type_general"`

### Expected behavior

Based on the docs, the introspection should pull the current connected database
and generate the models

### Prisma information

```
generator client {
  provider = "prisma-client-js"
  output = "client/nakama"
}

datasource cockroach {
  provider = "cockroachdb"
  url      = env("COCKROACH_DATABASE_URL")
}
```

### Environment & setup

- OS: macos
- Database: CockroachDB
- Node.js version: 20.9.0

### Prisma Version

```
5.7.1
```

**Comment by ludralph** [comment]
Hi @danieltamas 👋 

Thank you for raising this question. 

What version of Cockroach DB are you using so I can try reproducing using the exact same version?

**Comment by danieltamas** [comment]
Hi @ludralph thanks for being quick.
Here it is: `cockroachdb/cockroach:v19.1.5`

**Reply by janpio** [reply]
Prisma unfortunately can only support CockroachDB version 21.2.4+ (overview: https://www.prisma.io/docs/orm/reference/supported-databases)

**Reply by danieltamas** [reply]
I see. thanks @janpio any suggested solutions?
the reason: it's a live system and currently we're stuck if prisma can't connect

**Reply by janpio** [reply]
When you identify your database as CockroachDB, Prisma automatically runs a command that uses `enable_experimental_alter_column_type_general` ([see code here](https://github.com/prisma/prisma-engines/blob/0fb4ba094a61e322c9a21dd664c30f0bc36ee80d/schema-engine/connectors/sql-schema-connector/src/flavour/postgres.rs#L615-L617)) which does not seem to exist in the version you are using. Beyond that, there will probably be other functions used that are also not available in older versions, sorry.

**Reply by danieltamas** [reply]
Got it. Thank you @janpio

**Reply by danieltamas** [reply, marked as answer]
For anyone having the same conundrum, a workaround is to just manually create the models
since the client then works. Hope this helps.

Thanks
